src/input/return_enums.rs

```diff
@@ -147,12 +148,51 @@ macro_rules! any_next_error {
     };
 }
 
+macro_rules! unique_check {
+    ($py:expr, $input:expr, $field_type:expr, $unique:expr, $data:expr) => {
+        unique_check!($py, $input, $field_type, $unique, $data, item, item.as_ref($py));
+    };
+    ($py:expr, $input:expr, $field_type:expr, $unique:expr, $data:expr, $item:ident, $item_py:expr) => {
+        if $unique {
+            let repeated = if $data.len() <= 30 {
+                let mut hashes = Vec::with_capacity($data.len());
+                for $item in $data.iter() {
+                    hashes.push($item_py.hash()?);
+                }
+                (1..hashes.len()).any(|i| hashes[i..].contains(&hashes[i - 1]))
+            } else {
+                let mut set = AHashSet::with_capacity($data.len());
+                for $item in $data.iter() {
+                    set.insert($item_py.hash()?);
+                }
+                set.len() != $data.len()
```

Comment:
Hmm, as I mentioned on the PR comment, I think taking python's notion of uniqueness - which just compares hashes - would be an ok behavior. Otherwise, an hypothetical validation of `conset(unique=True)` over a value `[1, True]` would pass, but would return `{1}`, which just seems weird.

Reply:
That said, I could take type into consideration, which would result in the above example pass, but that would be weird.

Reply:
Code example demonstrating:
```python
class A:
    def __hash__(self):
        return 0


class B:
    def __hash__(self):
        return 0


my_dict = {A(): 1, B(): 2, 0: 3}

for k, v in my_dict.items():
    print(hash(k), k, v)

"""
0 <__main__.A object at 0x1052a1d90> 1
0 <__main__.B object at 0x1052a1dd0> 2
0 0 3
"""
```

Reply:
Ah, I see what you mean, yeah, my bad. I'll change the code to reflect that.

Reply:
Yes, I know, `__eq__` can also get overridden anyway. I thought the 'this is not safe' comment was mainly related to Python treating a few different types as equal (`True == 1`), but I get it now. I already have an idea of how I can fix this, I'll try to work on it later today.

Reply:
Btw, as far as I normally think of "uniqueness" in Python, `1` and `True` are equivalent, even if it seems a little counterintutive.

```python
>>> {1, True}
{1}
```

I would wager the only way this can be implemented correctly is by using Python hashing + equality.